peers/connect_peer.js

```diff
@@ -33,22 +35,41 @@ module.exports = ({id, lnd}, cbk) => {
         return cbk();
       },
 
+      // Get node
+      getNodeInfo: ['validate', ({}, cbk) => {
+        return getNode({lnd, is_omitting_channels: true, public_key: id}, (err, res) => {
+          // Ignore errors, node may not be found
+          if (!!err) {
+            return cbk();
+          }
+
+          return cbk(null, res);
+        });
+      }],
+
       // Get already connected peers
       getPeers: ['validate', ({}, cbk) => getPeers({lnd}, cbk)],
 
       // Get the node sockets if necessary to conneect
-      getSockets: ['getPeers', ({getPeers}, cbk) => {
+      getSockets: ['getNodeInfo', 'getPeers', ({getNodeInfo, getPeers}, cbk) => {
+        const isExistingPeer = getPeers.peers.map(n => n.public_key).includes(id);
+
         // Exit early when there is no need to connect to the node
-        if (getPeers.peers.map(n => n.public_key).includes(id)) {
+        if (!!isExistingPeer) {
           return cbk();
         }
 
-        return getNode({
-          lnd,
-          is_omitting_channels: true,
-          public_key: id,
-        },
-        cbk);
+        // Exit early when sockets are provided
+        if (!!sockets && !!isArray(sockets) && !!sockets.length) {
```

Comment:
if you do pass sockets, it should be validated to be a non empty array so you don't have to check here